doc/devel/development_workflow.rst

```diff
@@ -79,19 +79,13 @@ default, git will have a link to your fork of the GitHub repo, called
 
    git push origin my-new-feature
 
-In git >= 1.7 you can ensure that the link is correctly set by using the
-``--set-upstream`` option::
+.. hint::
 
-   git push --set-upstream origin my-new-feature
-
-From now on git will know that ``my-new-feature`` is related to the
-``my-new-feature`` branch in the GitHub repo.
-
-If you first opened the pull request from your ``main`` branch and then
-converted it to a feature branch, you will need to close the original pull
-request and open a new pull request from the renamed branch. See
-`GitHub: working with branches
-<https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.
+    If you first opened the pull request from your ``main`` branch and then
+    converted it to a feature branch, you will need to close the original pull
+    request and open a new pull request from the renamed branch. See
+    `GitHub: working with branches
+    <https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-branches#working-with-branches>`_.
```

Comment:
I know you just converted the text to a hint, which is an improvment. But reading this, it feels out of context. We're discussing making a new feature branch. Already having a PR from main means the user has gone two steps in a different direction already and it is not relevant to the feature branch.

I'm slightly in favor of removing this completely. Alternatively, if you think, this is still valuable (Do we expect users to find this when they need it? Is a PR from main really that bad that we need to discuss recovering from this non-standard workflow?), you can move this down to the section "Open a pull request".

Reply:
Ok let's get this in as an incremental improvement. We can discuss this separately.